migration/src/main/java/io/crate/migration/MigrationTool.java

```diff
@@ -0,0 +1,270 @@
+                                            tableName,
+                                            node);
+                                        summaryStats.addFailed(table, node);
+                                        continue;
+                                    }
+
+                                    if (IndexMetaDataUtil.checkAlreadyMigrated(shardDir)) {
+                                        LOGGER.debug("Shard [{}] in directory [{}] for {}table [{}] of node [{}] " +
+                                                     "is already migrated",
+                                            i,
+                                            shardIndexPath,
+                                            table.isPartitioned() ? "partitioned " : "",
+                                            tableName,
+                                            node);
+                                        continue;
+                                    }
+
+                                    if (!configuration.isDryrun()) {
+                                        LOGGER.debug("Migrating shard [{}] in directory [{}] for {}table [{}] " +
+                                                     "of node [{}]",
+                                            i,
+                                            shardIndexPath,
+                                            table.isPartitioned() ? "partitioned " : "",
+                                            tableName,
+                                            node);
+                                        new IndexUpgrader(shardDir, InfoStream.NO_OUTPUT, true).upgrade();
+                                        LOGGER.debug("Shard [{}] in directory [{}] for {}table [{}] of node [{}] " +
+                                                     "migrated successfully",
+                                            i,
+                                            shardIndexPath,
+                                            table.isPartitioned() ? "partitioned " : "",
+                                            tableName,
+                                            node);
+                                    }
+                                    if (!configuration.isDryrun()) {
+                                        summaryStats.addSuccessFull(table, node);
+                                    }
+                                }
+                            } catch (Exception e) {
+                                LOGGER.error("Error while upgrading {}table [{}] of node [{}]",
+                                    table.isPartitioned() ? "partitioned " : "",
+                                    tableName,
+                                    node,
+                                    e);
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (!configuration.isDryrun()) {
+                LOGGER.info(summaryStats.print());
```

Comment:
think it makes sense to always print a summary, if one does a dry-run, it's mandatory to know what would have happend